plugins/modules/remote_management/redfish/ilo_redfish_info.py

```diff
@@ -0,0 +1,191 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+###
+# Copyright (2021) Hewlett Packard Enterprise Development LP
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+###
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: ilo_redfish_info
+short_description: Gathers server information through iLO using Redfish APIs
+description:
+  - Builds Redfish URIs locally and sends them to iLO to
+    get information back.
+  - For use with HPE iLO operations that require Redfish OEM extensions.
+options:
+  category:
+    required: true
+    description:
+      - List of categories to execute on iLO.
+    type: list
+    elements: str
+  command:
+    required: true
+    description:
+      - List of commands to execute on iLO.
+    type: list
+    elements: str
+  baseuri:
+    required: true
+    description:
+      - Base URI of iLO.
+    type: str
+  username:
+    description:
+      - User for authentication with iLO.
+    type: str
+  password:
+    description:
+      - Password for authentication with iLO.
+    type: str
+  auth_token:
+    description:
+      - Security token for authentication with iLO.
+    type: str
+    version_added: 2.3.0
+  timeout:
+    description:
+      - Timeout in seconds for URL requests to iLO.
+    default: 10
+    type: int
+
+requirements:
+    - "python >= 3.8"
+    - "ansible >= 3.2"
+author:
+    - "Bhavya B (@bhavya06)"
+'''
+
+EXAMPLES = '''
+  - name: Get iLO Sessions
+    community.general.ilo_redfish_info:
+      category: Sessions
+      command: GetiLOSessions
+      baseuri: "{{ baseuri }}"
+      username: "{{ username }}"
+      password: "{{ password }}"
+    register: result_sessions
+'''
+RETURN = '''
+ilo_redfish_info:
+    description: Returns iLO sessions.
+    contains: 
```

Comment:
This contains a field `GetiLOSessions` (which is only there if that command is used), and that in turn contains fields like `msg` and `ret`. And probably some other dictionary that contains the information you claim is contained here.

Reply:
Made the changes.